torchgeo/datasets/cms_mangrove_canopy.py

```diff
@@ -0,0 +1,246 @@
+        "SouthAfrica",
+        "SriLanka",
+        "Sudan",
+        "Suriname",
+        "Taiwan",
+        "Tanzania",
+        "Thailand",
+        "TimorLeste",
+        "Togo",
+        "Tonga",
+        "TrinidadAndTobago",
+        "TurksAndCaicosIslands",
+        "Tuvalu",
+        "UnitedArabEmirates",
+        "UnitedStates",
+        "Vanuatu",
+        "Venezuela",
+        "Vietnam",
+        "VirginIslandsUs",
+        "WallisAndFutuna",
+        "Yemen",
+    ]
+
+    measurements = ["agb", "hba95", "hmax95"]
+
+    def __init__(
+        self,
+        root: str = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        measurement: str = "agb",
+        country: str = all_countries[0],
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        cache: bool = True,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            measurement: which of the three measurements, 'agb', 'hba95', or 'hmax95'
+            country: country for which to retrieve data
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            FileNotFoundError: if no files are found in ``root``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        self.root = root
+        self.checksum = checksum
+
+        assert isinstance(country, str), "Country argument must be a str."
+        assert (
+            country in self.all_countries
+        ), "You have selected an invalid country, please choose one of {}".format(
+            self.all_countries
+        )
+        self.country = country
+
+        assert isinstance(measurement, str), "Measurement must be a string."
+        assert (
+            measurement in self.measurements
+        ), "You have entered an invalid measurement, please choose one of {}.".format(
+            self.measurements
+        )
+        self.measurement = measurement
+
+        self.filename_glob = "Mangrove_{}_{}*".format(self.measurement, self.country)
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found or corrupted. "
+                "Please ensure you have manually downloaded it."
+            )
+
+        super().__init__(root, crs, res, transforms, cache)
+
+    def _check_integrity(self) -> bool:
```

Comment:
Can you follow the style of other datasets that use `_verify`? This will allow the dataset to work even if the user deletes the zip file and keeps only the single country/measurement file they need. See CDL for an example of this.